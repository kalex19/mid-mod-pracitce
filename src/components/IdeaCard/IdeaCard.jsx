import React from 'react';
import './IdeaCard.css';

export default function IdeaCard({ title, description, id, deleteIdeas }){
	return (
		<div>
			<h2>{title}</h2>
			<h5>{description}</h5>
			<button onClick={() => deleteIdeas(id)}>ðŸ—‘</button>
		</div>
	);
}
