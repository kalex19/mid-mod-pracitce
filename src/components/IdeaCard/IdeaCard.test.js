import {
  shallow
} from 'enzyme';
import React from 'react';
import IdeaCard from './IdeaCard';

describe('IdeaCard', () => {
      let wrapper;
      let mockDeleteIdeas;
      let mockIdeas;
      let mockKey;

      beforeEach = () => {
        mockDeleteIdeas = jest.fn();
        mockIdeas = [{
          title: 'hello',
          body: 'world'
        }, {
          title: 'goodbye',
          body: 'moon'
        }]
        mockKey = 2
        wrapper = shallow( < IdeaCard deleteIdeas = {
            mockDeleteIdeas
          } {
            ...mockIdeas
          }
          key = {
            mockKey
          }
          /> )
        }

        it('should match snapshot', () => {
          expect(wrapper).toMatchSnapshot();
        })
      })