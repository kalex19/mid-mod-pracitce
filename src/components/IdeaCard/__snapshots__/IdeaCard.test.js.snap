// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard should match snapshot 1`] = `undefined`;
